source/features/comment-shortcuts.tsx

```diff
@@ -0,0 +1,79 @@
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+const shortcutClass = new Map<string, string>([
+	['e', '.rgh-edit-comment, [aria-label="Edit comment"]'],
+	['d', '[aria-label="Delete comment"]'],
+	['h', '[aria-label="Hide comment"]'],
+	['j', ''],
+	['k', '']
+]);
+
+function commentHash(comment: HTMLElement): string {
```

Comment:
This function feels like it should not be necessary.

`items` should only includes directly-focusable elements, not its parents/ancestors, so the right hash should always be just `chosen.hash`

Reply:
![image](https://user-images.githubusercontent.com/16872793/85015855-b8807c80-b136-11ea-8d2b-c268fd364f21.png)

The issue is we need number 1 but need to focus number 2. This is the only case we need it for

Reply:
Why do you _need_ number 1?

I think you only need exactly what you're about to focus.

Reply:
That is where the hash of the comment is

Reply:
Regular comments and review comments work one way requested changes and approve work a drop different

![image](https://user-images.githubusercontent.com/16872793/85071657-d6c19900-b185-11ea-9879-fb06dae2c318.png)

Reply:
I checked, your 1/2 comment was incorrect. 

1. Visit https://github.com/sindresorhus/refined-github/pull/3242#pullrequestreview-433511748
2. Run `$(':target')` in the console.

Neither the 1 nor the 2 are the `:target`, so you're selecting the wrong `items` below

Reply:
`a.js-timestamp` is a completely unrelated element and should not be selected.

What this needs to select is the focusable element itself, **not** a link that points to it.

There can be 10 `a.js-timestamp` that point to the same element and they're irrelevant to navigation.

Reply:
Let me try to explain. I spent a really long time trying to find unique thing to each comment and it should be easy to go to it.

From what I found it would be the comment time, since that has a hash. That hash takes us to the comment. **except for** requested changes and approve. those take us to the top part not the comment it self. We want to focus the comment.

I hope I am clear enough

Reply:
Let me try to explain. I spent a really long time trying to find unique thing to each comment and it should be easy to go to it.

However the comment hash on a requested changes and approve is above the comment

Reply:
i am dead wrong here. i had one long day. Sorry